src/components/taskDetails/ProgressUpdateCard/ProgressUpdateCardPresentation.tsx

```diff
@@ -1,60 +1,124 @@
 import React, { MouseEvent } from 'react';
-import { FaRegClock } from 'react-icons/fa6';
+import { FaAngleRight, FaRegClock } from 'react-icons/fa6';
 import Tooltip from '@/components/common/Tooltip/Tooltip';
 import styles from './progress-update-card.module.scss';
 
 type ProgressUpdateCardPresentationProps = {
     titleToShow: string;
-    isTitleWrapped: boolean;
-    isMoreButtonVisible: boolean;
-    onMoreButtonClick: (event: MouseEvent<HTMLElement>) => void;
     onHoverOnDate: (event: MouseEvent<HTMLElement>) => void;
     onMouseOutOnDate: (event: MouseEvent<HTMLElement>) => void;
     dateInAgoFormat: string;
     isTooltipVisible: boolean;
     tooltipString: string;
+    dataToShowState: ProgressUpdateDataToShow[];
+    onMoreOrLessButtonClick: (
+        e: MouseEvent<HTMLElement>,
+        clickedOnData: ProgressUpdateDataToShow
+    ) => void;
+    onCardClick: (event: MouseEvent<HTMLElement>) => void;
+    isExpanded: boolean;
 };
+
+type ProgressUpdateDataToShow = {
+    id: string;
+    label: string;
+    body: string;
+    trimmedBody: string;
+    shouldReadMoreButtonShow: boolean;
+    isReadMoreEnabled: boolean;
+};
+
 export default function ProgressUpdateCardPresentation({
     titleToShow,
-    isTitleWrapped,
-    isMoreButtonVisible,
-    onMoreButtonClick,
     onHoverOnDate,
     onMouseOutOnDate,
     dateInAgoFormat,
     isTooltipVisible,
     tooltipString,
+    dataToShowState,
+    onMoreOrLessButtonClick,
+    onCardClick,
+    isExpanded,
 }: ProgressUpdateCardPresentationProps) {
-    return (
-        <div className={styles['progress-update-card']}>
-            <h3 className={styles['progress-update-card__title']}>
-                {titleToShow}
-                {isMoreButtonVisible && (
-                    <button
-                        className={styles['progress-update-card__more-button']}
-                        onClick={onMoreButtonClick}
-                    >
-                        {isTitleWrapped ? 'More' : 'Less'}
-                    </button>
-                )}
-            </h3>
-            <span
-                className={styles['progress-update-card__date-container']}
-                onMouseOver={onHoverOnDate}
-                onMouseOut={onMouseOutOnDate}
+    const progressInfoMapping = dataToShowState.map(
+        (datum: ProgressUpdateDataToShow) => (
```

Comment:
Not sure what variable name means.

Reply:
it's mapping of progress updates info

Reply:
`datum` gives me no context tbh.

Reply:
datum is a singular form of data, it's only used in the parameter here and also I have added types of that datum